src/Controller/Component/CsrfComponent.php

```diff
@@ -56,7 +56,7 @@ class CsrfComponent extends Component
         'cookieName' => 'csrfToken',
         'expiry' => 0,
         'secure' => false,
-        'httpOnly' => false,
+        'httpOnly' => true,
```

Comment:
Question is, do we really need a CSRF hidden input field or won't a submitted cookie be enough?

When looking at what my Forms actually do generate. Hidden fields with _Token[fields] and _Token[unlocked].
Only when adding CSRFComponent I get <input type="hidden" name="_csrfToken" value="...."/>
What's the point of that if a cookie can be read.

I'd say CSRF-protection should work two ways:
a) httpOnly cookie (which can be changed)
b) http header and html meta tag - which can be read by javascript even if you are not using cake bake form helper

Why should XSS attacks be only relevant for the same domain? An XSS attack could read the CSRF key and send it over the wire via javascript to some other server which then sets up a prepared cross site request forgery attack to be executed by some user joe?